framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -736,8 +736,11 @@ protected void addInformationFromPreconditions(
                         } else {
                             thenStore.insertValue(secondInternal, firstValue);
                         }
-                        return new ConditionalTransferResult<>(
-                                res.getResultValue(), thenStore, elseStore);
+                        // To handle `(a = b = c) == x`, repeat for all insertable receivers of
+                        // splitted assignments instead of returning.
```

Comment:
Was there already a test for this and it failed without this change?
If not, can you add a dedicated test that triggers this problem?